src/main/java/seedu/duke/parsers/AddParser.java

```diff
@@ -38,16 +41,31 @@ public AddParser() {
         groupNames.add(TASK_WORKING_TIME);
         groupNames.add(MODULE_CODE);
         groupNames.add(MODULE_DESCRIPTION);
+        groupNames.add(TASK_DESCRIPTION_TWO);
+        groupNames.add(INVALID);
     }
 
     @Override
     public Command parseCommand(String userInput) throws ModHappyException {
         HashMap<String, String> parsedArguments = parseString(userInput);
         final String taskName = parsedArguments.get(TASK_NAME);
-        final String taskDescription = parsedArguments.get(TASK_DESCRIPTION);
         final String estimatedWorkingTime = parsedArguments.get(TASK_WORKING_TIME);
         final String moduleCode = parsedArguments.get(MODULE_CODE);
         final String moduleDescription = parsedArguments.get(MODULE_DESCRIPTION);
+        final String taskDescriptionTwo = parsedArguments.get(TASK_DESCRIPTION_TWO);
+        String invalid = parsedArguments.get(INVALID);
+        String taskDescription = parsedArguments.get(TASK_DESCRIPTION);
+        if (invalid.isEmpty()) {
+            invalid = NULL_FIELD;
+        }
+        boolean isInvalid = ((!Objects.equals(invalid, NULL_FIELD))
+                || ((!Objects.equals(taskDescription, NULL_FIELD))
+                && (!Objects.equals(taskDescriptionTwo, NULL_FIELD))));
```

Comment:
Perhaps Objects.isNull(object) can replace Objects.equals if just checking whether the object is null. Also, not directed only at the usage here, but is there a need to assign null to a constant?

Reply:
changed to isNull and NULL_FIELD to just null